apps/web/src/pages/ApplicantDashboardPage/components/FunctionalCommunityListItem.tsx

```diff
@@ -0,0 +1,196 @@
+    <span>
+      <BriefcaseIcon
+        data-h2-color="base(success)"
+        {...sharedIconStyling}
+      ></BriefcaseIcon>
+      {intl.formatMessage({
+        defaultMessage: "Interested in work",
+        id: "1VKNrs",
+        description: "Phrase marking interest in community work opportunities",
+      })}
+    </span>
+  );
+  const notInterestedWork = (
+    <span>
+      <XCircleIcon
+        data-h2-color="base(gray.lighter)"
+        {...sharedIconStyling}
+      ></XCircleIcon>
+      {intl.formatMessage({
+        defaultMessage: "Not interested in work",
+        id: "VDVRPt",
+        description:
+          "Phrase marking lack of interest in community work opportunities",
+      })}
+    </span>
+  );
+  const missingWork = (
+    <span>
+      <ExclamationTriangleIcon
+        data-h2-color="base(error)"
+        {...sharedIconStyling}
+      ></ExclamationTriangleIcon>
+      {intl.formatMessage({
+        defaultMessage: "<red>Missing work info</red>",
+        id: "ZrLeP8",
+        description:
+          "Phrase marking incomplete community work opportunities interest",
+      })}
+    </span>
+  );
+
+  const generateMetaDataJobInterest = (
+    jobInterest: boolean | null | undefined,
+  ): JSX.Element => {
+    if (jobInterest === null || jobInterest === undefined) {
+      return missingWork;
+    }
+    return jobInterest ? interestedWork : notInterestedWork;
+  };
+
+  const interestedTraining = (
+    <span>
+      <PresentationChartLineIcon
+        data-h2-color="base(success)"
+        {...sharedIconStyling}
+      ></PresentationChartLineIcon>
+      {intl.formatMessage({
+        defaultMessage: "Interested in training",
+        id: "ERsZAD",
+        description:
+          "Phrase marking interest in community training opportunities",
+      })}
+    </span>
+  );
+  const notInterestedTraining = (
+    <span>
+      <XCircleIcon
+        data-h2-color="base(gray.lighter)"
+        {...sharedIconStyling}
+      ></XCircleIcon>
+      {intl.formatMessage({
+        defaultMessage: "Not interested in training",
+        id: "8wU0cq",
+        description:
+          "Phrase marking lack of interest in community training opportunities",
+      })}
+    </span>
+  );
+  const missingTraining = (
+    <span>
+      <ExclamationTriangleIcon
+        data-h2-color="base(error)"
+        {...sharedIconStyling}
+      ></ExclamationTriangleIcon>
+      {intl.formatMessage({
+        defaultMessage: "<red>Missing training info</red>",
+        id: "bupAcV",
+        description:
+          "Phrase marking incomplete community training opportunities interest",
+      })}
+    </span>
+  );
+
+  const generateMetaDataTrainingInterest = (
+    trainingInterest: boolean | null | undefined,
+  ): JSX.Element => {
+    if (trainingInterest === null || trainingInterest === undefined) {
+      return missingTraining;
+    }
+    return trainingInterest ? interestedTraining : notInterestedTraining;
+  };
```

Comment:
We should avoid defining components inside of other components. Can this be extracted out into its own component?

https://react.dev/learn/your-first-component#nesting-and-organizing-components

Reply:
Hmm, I wasn't sure if a few spans counted as something worth factoring out into a separate component. I did it this way to avoid a nested ternary structure

Reply:
> [Components can render other components, but **you must never nest their definitions:**](https://react.dev/learn/your-first-component#nesting-and-organizing-components:~:text=Components%20can%20render%20other%20components%2C%20but%20you%20must%20never%20nest%20their%20definitions%3A)

They even added `<strong>` tags for emphasis 😆.

Reply:
Well, it requires `useIntl()` for each so I can't play it too casually with the functions. Nested ternary not a good idea.
Following filename casing of files like `cells.tsx`, I extracted out an icons file

https://github.com/GCTC-NTGC/gc-digital-talent/pull/12607/commits/fc1ec0393456753e0e96d4795715fb821cb1cdc5